banno-enrollment-po/BANNO.ENROLLMENT.V1.POW

```diff
@@ -0,0 +1,772 @@
+  END
+ DIALOGPROMPTLISTOPTION(ACTIONEXITPGM,"Exit Banno Enrollment")
+ DIALOGPROMPTLISTEND
+ DIALOGDISPLAY
+ ACTIONREQUESTED=ENTERNUMBER("Choose Action:",ACTIONCHGPASS)
+ DIALOGCLOSE
+END [PROCEDURE]
+
+PROCEDURE CHGPASSDIALOG
+[* Dialog that prompts user for new HB password and password
+** probation preference.
+*]
+ DIALOGSTART("Banno - New Password",200%,0)
+ DIALOGINTROTEXT("Changing Password for User "+PREFUSER(SELECTEDPREF))
+ DIALOGPROMPTCHAR("Enter New Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),"")
+ DIALOGPROMPTYESNO("Change Password at next login?",FALSE)
+ DIALOGDISPLAY
+ NEWPASSWORD=ENTERCHARACTER("Enter New Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),"")
+ NEWPASSPROB=ENTERYESNO("Change Password at next login?","0")
+ DIALOGCLOSE
+END [PROCEDURE]
+
+PROCEDURE CHGPASSFM
+[* File maintenance and error handling for password change.
+*]
+ IF NEWPASSPROB="Y" THEN
+  NEWPASSPROBNUM=1
+ ELSE
+  NEWPASSPROBNUM=0
+
+ FMPERFORM REVISE PREFERENCE LOC PREFLOC(SELECTEDPREF)(CHECKPRIVS,0,FMERROR)
+  DO
+   SET HBPASSWORD TO NEWPASSWORD
+   SET LASTHBPWCHANGEDATE TO SYSTEMDATE
+   SET PWDRESETPROBATION TO NEWPASSPROBNUM
+  END
+ IF FMERROR<>"" THEN
+  CALL ERRORHANDLER
+ ELSE IF FMERROR="" AND NEWPASSWORD<>"" THEN
+  POPUPMESSAGE(0,"Password Successfully Updated")
+ ELSE IF FMERROR="" AND NEWPASSWORD="" THEN
+  POPUPMESSAGE(0,"Password Successfully Blanked Out")
+END [PROCEDURE]
+
+PROCEDURE CREATEEDITDIALOG
+[* Dialog that prompts user for information needed when creating
+** or editing a user.
+*]
+ DIALOGSTART("Create/Edit Banno User",200%,1)
+ IF NEWEMAILADDR<>"" THEN
+  DIALOGPROMPTCHAR("Email : ",40,NEWEMAILADDR)
+ ELSE
+  DIALOGPROMPTCHAR("Email : ",40,PREFEMAILADDR(SELECTEDPREF))
+ IF NEWUSERNAME<>"" THEN
+  DIALOGPROMPTCHAR("User Name : ",GETDATANUMBER(GETPARAMHBUSERNAMEMAXLENGTH),NEWUSERNAME)
+ ELSE
+  DIALOGPROMPTCHAR("User Name : ",GETDATANUMBER(GETPARAMHBUSERNAMEMAXLENGTH),PREFUSER(SELECTEDPREF))
+ IF ACTIONREQUESTED=ACTIONCREATEUSR THEN
+  DO
+   IF NEWPASSWORD<>"" THEN
+    DIALOGPROMPTCHAR("Enter Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),NEWPASSWORD)
+   ELSE
+    DIALOGPROMPTCHAR("Enter Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),"")
+  END
+ ELSE IF ACTIONREQUESTED=ACTIONEDITUSER  THEN
+  DIALOGINTROTEXT("Use Change Password Option to change Password")
+ IF ACTIONREQUESTED=ACTIONEDITUSER AND PREFPASSPROB(SELECTEDPREF)=0 THEN
+  DIALOGPROMPTYESNO("Change Password at next login?",FALSE)
+ ELSE IF ((ACTIONREQUESTED=ACTIONEDITUSER AND PREFPASSPROB(SELECTEDPREF)=1) OR
+           ACTIONREQUESTED=ACTIONCREATEUSR) THEN
+  DIALOGPROMPTYESNO("Change Password at next login?",TRUE)
+ IF ALLOWXFERTOANY=TRUE AND PREFXFERANYLOC(SELECTEDPREF)<>0 OR ACTIONREQUESTED=ACTIONCREATEUSR THEN
+  DIALOGPROMPTYESNO("Enable Xfer To Any Acct?",TRUE)
+ ELSE IF ALLOWXFERTOANY=TRUE AND PREFXFERANYLOC(SELECTEDPREF)=0 THEN
+  DIALOGPROMPTYESNO("Enable Xfer To Any Acct?",FALSE)
+ DIALOGDISPLAY
```

Comment:
Lines 398-399: Do these need to be in the same order as the corresponding DIALOGPROMPTCHAR statements? Can't remember if that's critical or just SOP

Reply:
They do not need to be in the same order, but I put them in order for consistency